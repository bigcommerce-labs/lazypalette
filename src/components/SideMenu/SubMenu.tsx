import React, { Component } from 'react';
import styled from 'styled-components';

import { activeClassName, NavItem } from './styles';
import MenuItems from './MenuItems';

const BackLink = styled.div`
  color: #34343B;
  font-size: 24px;
  margin-bottom: 30px;
  margin-left: 28px;
`;

interface SubMenuItem {
  label: string;
  path: string;
}

interface SubMenuProps {
  title: string;
  items?: SubMenuItem[];
  match: any;
}

class SubMenu extends Component<SubMenuProps, any> {
  static defaultProps: Partial<SubMenuProps> = {
    items: [],
  };

  constructor(props: SubMenuProps) {
    super(props);
  }

  render() {
    return (
      <div>
        <BackLink>
          <NavItem to="/" exact activeClassName={activeClassName}>&lt; {this.props.title}</NavItem>
        </BackLink>
        <MenuItems items={this.props.items!} match={this.props.match}/>
      </div>
    );
  }
}

export default SubMenu;
