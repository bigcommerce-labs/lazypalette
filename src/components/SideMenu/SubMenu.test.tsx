import { shallow } from 'enzyme';
import React from 'react';

import SubMenu from './SubMenu';

it('renders', () => {
    const testItems = [
        {
            label: 'Logo and name',
            path: 'logo',
        },
        {
            label: 'Store theme',
            path: 'theme',
        },
    ];

    const subMenu = shallow(
        <SubMenu title="Menu" currentPath="/" items={testItems}/>
    );

    expect(subMenu).toMatchSnapshot();
});
