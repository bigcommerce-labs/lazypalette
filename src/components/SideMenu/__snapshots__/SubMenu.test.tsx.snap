// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <styled.div>
    <Styled(NavLink)
      activeClassName="nav-item-active"
      exact={true}
      to="/"
    >
      <styled.span>
        <Icon
          glyph="chevronLeft"
          size="larger"
        />
      </styled.span>
      Menu
    </Styled(NavLink)>
  </styled.div>
  <withRouter(MenuItems)
    currentPath="/"
    items={
      Array [
        Object {
          "label": "Logo and name",
          "path": "logo",
        },
        Object {
          "label": "Store theme",
          "path": "theme",
        },
      ]
    }
  />
</div>
`;
